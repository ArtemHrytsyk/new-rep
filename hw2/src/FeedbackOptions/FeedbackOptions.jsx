import React from "react";

function FeedbackOptions({ goodFeedback, neutralFeedback, badFeedback }) {
    return (
        <React.Fragment>
            <button onClick={goodFeedback} className="button-good">Good:</button>
            <button onClick={neutralFeedback} className="button-neutral">Neutral:</button>
            <button onClick={badFeedback} className="button-bad">Bad:</button>
        </React.Fragment>
    );
}

export default FeedbackOptions;