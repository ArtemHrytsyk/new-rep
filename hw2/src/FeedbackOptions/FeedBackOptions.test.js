import React from 'react';
import FeedbackOptions from '../FeedbackOptions';
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
Enzyme.configure({ adapter: new Adapter() })
import { render, screen } from '@testing-library/react';

import '@testing-library/jest-dom'

const setUp = () => shallow(<FeedbackOptions />);

describe("FeedbackOptions (Buttons) component", () => {
    let component;
    let instance;
    beforeEach(() => {
        component = setUp();
        instance = component.instance();
    });

    it("Should render FeedbackOptions (Buttons) component", () => {
        expect(component).toMatchSnapshot();
    });

    describe('Test Button component', () => {
        it('Test click event', () => {
          const mockCallBack = jest.fn();
      
          const button = shallow((<FeedbackOptions onClick={mockCallBack}></FeedbackOptions>));
          button.find('.button-good').simulate('click');
          expect(mockCallBack.mock.calls.length).toEqual(1);
        });
      });

      
});