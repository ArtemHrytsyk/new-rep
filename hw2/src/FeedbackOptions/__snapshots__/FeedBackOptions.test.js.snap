// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackOptions (Buttons) component Should render FeedbackOptions (Buttons) component 1`] = `ShallowWrapper {}`;
